packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -632,14 +658,25 @@ mixin _CupertinoSheetRouteTransitionMixin<T> on PageRoute<T> {
   Duration get transitionDuration => const Duration(milliseconds: 500);
 
   @override
-  DelegatedTransitionBuilder? get delegatedTransition =>
-      CupertinoSheetTransition.delegateTransition;
+  DelegatedTransitionBuilder? get delegatedTransition {
+    if (hasCustomTopGap) {
+      return null;
+    }
+    return CupertinoSheetTransition.delegateTransition;
+  }
 
   /// Determines whether the content can be dragged.
   ///
   /// If `true`, dragging is enabled; otherwise, it remains fixed.
   bool get enableDrag;
 
+  /// The gap between the top of the screen and the top of the sheet as a ratio
+  /// of the screen height (0.0 to 1.0). Defaults to [_kTopGapRatio].
```

Comment:
Private values are not accessibile by documentation. 
```suggestion
  /// of the screen height (0.0 to 1.0). Defaults to a value of 0.08.
```